report/23_Process-CI-CD-Description.md

```diff
@@ -1,2 +1,13 @@
-A complete description of stages and tools included in the CI/CD chains.
-That is, including deployment and release of your systems.
+All CI/CD in the project is achieved using GitHub actions. The different steps of the CI/CD process have been split into workflow files with separate responsibilities:
+- ```dotnetcore.yml``` runs unit tests on the project whenever code is pushed.
+- ```e2e.yml``` runs end-to-end tests when a pull request to master is made.
+- ```sonarPR.yml``` provides a report with code analysis from SonarCloud whenever a pull request to master is made. This report is put on the pull request for easy access.
+- ```sonarMaster.yml``` is triggered when code is pushed to master, and updates the general [SonarCloud report](https://sonarcloud.io/project/issues?id=jlndk_devoops) for the project. This is arguably not part of the CI/CD flow. 
+- ```deploy.yml``` deploys the code on the Digital Ocean server whenever code is pushed to master.
+- ```release.yml``` makes a release of the code, including automatic versioning and a copy of this report, whenever code is pushed to master
+
+Continous Integration is accomplished through ```dotnetcore.yml``` and ```e2e.yml``` by testing code to make sure that it doesn't break when the project changes. ```dotnetcore.yml``` is triggered on all pushes to speed up testing, and so a developer can check that their code still works before they make a pull request. Ideally ```e2e.yml``` would also be run on pushes, but the action is relatively resource-intensive so it was decided to only run it when a developer is sure that their code is ready for master. The same is the case for the ```sonarPR.yml```. 
```

Comment:
The resource intensiveness is a non issue. We don't pay for it, the issues is that it would clutter the actions, especially as the e2e takes a long time to run. The e2e also realises on there being a PR it can comment on, without this it fails.